tests/cases/conformance/types/mapped/mappedTypeRelationships.ts

```diff
@@ -192,3 +192,21 @@ function f90<T extends { x: number }>() {
 function f<T extends { x: {} }>(): Partial<T> {
     return undefined! as T;
 }
+
+// #32365
+interface SettingsTypes {
+  audio: {
+    volume: string;
+  };
+  video: {
+    resulution: string;
+  };
+}
+interface Settings<Params extends { [K in keyof Params]?: string }> {
+  config: Params;
+}
+type ConcreteSettingsResult1 = Settings<SettingsTypes["audio"]>;
+type ConcreteSettingsResult2 = Settings<SettingsTypes["audio" | 'video']>;

```

Comment:
Let's use a single quote style here.

Reply:
Should one read this as "(single quote) style" or "single (quote style)" 🚎

Reply:
The latter, e.g. just use double quotes.